lib/r10k/source/hg.rb

```diff
@@ -0,0 +1,124 @@
+require 'r10k/hg'
+require 'r10k/hg/cache'
+require 'r10k/environment/hg'
+require 'r10k/environment/name'
+
+# This class implements a source for Git environments.
+#
+# A Git source generates environments by locally caching the given Git
+# repository and enumerating the branches for the Git repository. Branches
+# are mapped to environments without modification.
+#
+# @since 1.3.0
+class R10K::Source::Hg < R10K::Source::Base
+
+  include R10K::Logging
+
+  R10K::Source.register(:hg, self)
+
+  # @!attribute [r] type
+  #   @return [:hg] The type of this control repo source
+  attr_reader :type
```

Comment:
I had accidentally made these changes on the r10k-api branch and that's a leftover from that branch. I'll get rid of this.